src/classes/Households.cls

```diff
@@ -130,18 +141,278 @@ public inherited sharing class Households extends fflib_SObjects {
+            householdNamingServiceInst.updateHouseholdNameAndMemberCountAsync(
                     new List<Id>(ids()));
         } else {
-            HouseholdNamingService householdNamingService = new HouseholdNamingService();
-            householdNamingService.updateHouseholdNameAndMemberCount(new List<Id>(ids()));
+            householdNamingServiceInst.updateHouseholdNameAndMemberCount(new List<Id>(ids()));
         }
     }
 
     public Set<Id> ids() {
         return getIdFieldValues(npo02__Household__c.Id);
     }
 
+    /*******************************************************************************************************
+    * @description Updates Contacts to relate to a single Individual Account
+    * @param contacts the list of Contacts that need updating.
+    */
+    public static void attachToBucketAccount(List<Contact> contactsWithoutBucketAccount){
+        if (contactsWithoutBucketAccount.isEmpty()) {
+            return;
+        }
+
+        Id individualAccountId = CAO_Constants.getIndividualAccountId();
+
+        if (individualAccountId != null) {
+            for (Contact contactRecord : contactsWithoutBucketAccount) {
+                //connect all Contacts to the bucket Account
+                contactRecord.AccountId = individualAccountId;
+            }
+        } else {
+            //if there is no bucket Account, the bucket model won't function unless we create one
+            Account individual = new Account();
+            individual.Name = CAO_Constants.BUCKET_ACCOUNT_NAME;
+            //flag the account as an individual account
+            individual.npe01__SYSTEMIsIndividual__c = true;
+            //individual.Type = CAO_Constants.BUCKET_ORGANIZATION_TYPE;
+            individual.npe01__SYSTEM_AccountType__c = CAO_Constants.BUCKET_ORGANIZATION_TYPE;
+            UTIL_DMLService.insertRecord(individual);
+
+            for (Contact contactRecord : contactsWithoutBucketAccount) {
+                //connect Contact to bucket Account
+                contactRecord.AccountId = individual.Id;
+            }
+        }
+    }
+
+    /**
+     * @description Determines whether or not the contact needs a one to one account assigned.
+     * @param  The contact record to verify.
+     * @param  The existing account record assocatied with the contact.
+     * @return Boolean - true when the contact needs a one to one account.
+     */
+    //todo: fix
+    public static Boolean needsOneToOneAccount(Contact contactRecord, Account accountRecord) {
+        return contactRecord.npe01__Private__c != true
+                && (contactRecord.AccountId == null
+                || (accountRecord != null
+                && accountRecord.npe01__SYSTEM_AccountType__c == CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE));
+    }
+
+    public static List<Contact> contactsWithoutAccounts(Contacts contactsInstance, Map<Id, Contact> oldContactById) {
```

Comment:
```suggestion
    public static List<Contact> contactsWithoutAccounts(Contacts contactsInstance) {
```

`oldContactById` is equivalent to `contactInstance.oldMap` here I think, so we could drop the second argument.